From: Josef Bacik <jbacik@fusionio.com>
Date: Mon, 22 Oct 2012 15:51:44 -0400
Patch-mainline: 3.7
Git-commit: c37b2b6269ee4637fb7cdb5da0d1e47215d57ce2
References: FATE#312888
Subject: [PATCH] Btrfs: do not bug when we fail to commit the
 transaction

We BUG if we fail to commit the transaction when creating a snapshot, which
is just obnoxious.  Remove the BUG_ON().  Thanks,

Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -578,7 +578,8 @@ static int create_snapshot(struct btrfs_
 		ret = btrfs_commit_transaction(trans,
 					       root->fs_info->extent_root);
 	}
-	BUG_ON(ret);
+	if (ret)
+		goto fail;
 
 	ret = pending_snapshot->error;
 	if (ret)
